tests/tests/tier0/bluechi-is-online-system/test_bluechi_is_online_system.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+import logging
+from typing import Dict
+
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+
+LOGGER = logging.getLogger(__name__)
+
+AGENT_ONE = "agent-one"
+AGENT_TWO = "agent-two"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    agent_one = nodes[AGENT_ONE]
+    agent_two = nodes[AGENT_TWO]
+
+    # Test 1: Both agents and controller are online
+    LOGGER.debug("Testing with both agents and controller online.")
+    result, output = ctrl.exec_run("bluechi-is-online system")
+    assert result == 0, "Expected successful execution of bluechi-is-online system"
+    assert not output, "Expected no output when all agents are online"
+
+    # Test 2: Agent-one offline
+    LOGGER.debug("Stopping agent-one.")
+    agent_one.systemctl.stop_unit("bluechi-agent")
+    assert agent_one.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+
+    result, output = ctrl.exec_run("bluechi-is-online system")
+    assert (
+        result != 0
+    ), "Expected failure of bluechi-is-online system when agent-one is offline"
+
+    # Bring agent-one back online
+    LOGGER.debug("Starting agent-one.")
+    agent_one.systemctl.start_unit("bluechi-agent")
+    assert agent_one.wait_for_unit_state_to_be("bluechi-agent", "active")
+
+    # Test 3: Agent-two offline
```

Comment:
Wouldn't it be better to test all agents offline scenario instead of the current Test 3, which tests pretty much the same scenario as Test 2 (one agent offline) ?

Reply:
I think it's a good test because we test each time different agent is offline, so we verify 'bluechi-is-online system' is checking agent by agent (before returning 0) and not only the first one.
I can add a test for checking both agents are offline.
what do you think?

Reply:
I'm not sure it's worth, the condition is "at least one agent offline", so if using just first one is too strict, then use random numbers to select first or second. But adding all offline seems to me more important then checking first agent offline and then second agent offline.

@engelmi @mkemel WDYT?

Reply:
I think both agents offline is important, but adding or replacing existing check is about the same effort, so it's worth to have both

Reply:
> Because 'bluechi-is-online system' returns 0 or 1 without additional output, nothing guarantees that the command doesn't return 0 when it encounters the first agent that is online.

In that case you'd need to check `bluechi-is-online system` returns 0 after bringing `agent-one` back up, otherwise the check might be still related to the first agent going offline, right?

Personally, I don't think its necessary to check for each agent individually. We already have [monitor-system-status](https://github.com/eclipse-bluechi/bluechi/tree/main/tests/tests/tier0/monitor-system-status) test to ensure the online, degraded, offline status is properly updated here - and bluechi-is-online hooks into that. But from a Black Box Testing perspective it can make sense to test this. So up to you, I'd say. 
However, please also check the case where all agents are offline.

Reply:
Done. I kept agent-two is offline and added another test for both agents are offline.